atc/api/pipelineserver/rename.go

```diff
@@ -28,13 +29,39 @@ func (s *Server) RenamePipeline(pipeline db.Pipeline) http.Handler {
 			return
 		}
 
+		var warnings []atc.ConfigWarning
+		err = atc.ValidateIdentifier(rename.NewName, "pipeline")
+		if err != nil {
+			warnings = append(warnings, atc.ConfigWarning{
```

Comment:
I think we should error explicitly here with a 400 Bad Request. I think the idea of the warnings is to remove the friction when existing pipelines already have invalid identifiers, but we should prevent renaming to an invalid identifier